packages/flutter/lib/src/material/stepper.dart

```diff
@@ -938,3 +999,167 @@ class _TrianglePainter extends CustomPainter {
     );
   }
 }
+
+/// The `StepStyle` class provides customization options for configuring the appearance of a step in a stepper widget.
+///
+/// To create a customized step style, instantiate a `StepStyle` object and pass it to the `Step` widget.
+///
+/// Example usage:
+/// ```dart
+/// Step(
+///   title: const Text('Step 1'),
+///   content: const Text('Content for Step 1'),
+///   stepStyle: StepStyle(
+///     color: Colors.blue,
+///     errorColor: Colors.red,
+///     border: Border.all(color: Colors.grey),
+///     boxShadow: const BoxShadow(blurRadius: 3.0, color: Colors.black26),
+///     gradient: const LinearGradient(colors: <Color>[Colors.red, Colors.blue]),
+///     indexStyle: const TextStyle(color: Colors.white),
+///   ),
+/// )
+/// ```
+/// See also:
+/// {@tool dartpad}
+/// An example the shows how to use the [StepStyle], and the [StepStyle] UI
+/// appearance.
+///
+/// ** See code in examples/api/lib/material/stepper/step_style.0.dart **
+/// {@end-tool}
+
+@immutable
+class StepStyle with Diagnosticable {
+  /// Constructs a [StepStyle].
+  const StepStyle({
+    this.color,
+    this.errorColor,
+    this.connectorColor,
+    this.connectorThickness,
+    this.border,
+    this.boxShadow,
+    this.gradient,
+    this.indexStyle,
+  });
+
+  /// The color of the circle in the step.
```

Comment:
```suggestion
  /// Overrides the default color of the circle in the step.
```

Similar feedback for the other StepStyle properties.